libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/api/ElasticsearchEntitlementChecker.java

```diff
@@ -1322,4 +1325,40 @@ public void checkName(Class<?> callerClass, FileStore that) {
     public void checkType(Class<?> callerClass, FileStore that) {
         policyManager.checkReadStoreAttributes(callerClass);
     }
+
+    @Override
+    public void checkPathToRealPath(Class<?> callerClass, Path that, LinkOption... options) {
+        if (EntitlementChecker.class.isAssignableFrom(callerClass)) {
```

Comment:
What is this for?

Reply:
That's to prevent above mentioned stackoverflow due to having the checker invoke toRealPath itself... if we all agree, I'll remove 👍